torchvision/csrc/ops/cuda/interpolate_aa_kernels.cu

```diff
@@ -0,0 +1,396 @@
+          w2,
+          width1,
+          rwidth,
+          support_w,
+          wx,
+          interp_width,
+          bicubic_filter,
+          xmin,
+          xsize);
+      _compute_weights<scalar_t, accscalar_t, filter_fn_t>(
+          h2,
+          height1,
+          rheight,
+          support_h,
+          wy,
+          interp_height,
+          bicubic_filter,
+          ymin,
+          ysize);
+    }
+
+    for (int n = 0; n < batchsize; n++) {
+      for (int c = 0; c < channels; ++c) {
+        // interpolate on x-axis for ymin to ymin + ysize
+        for (int64_t y = 0; y < ysize; y++) {
+          // copy data into the local buffer and use
+          // interpolate_aa_single_dim method
+          for (int x = 0; x < xsize; x++) {
+            buffer1[x] = idata[n][c][ymin + y][xmin + x];
+          }
+
+          buffer2[y] = static_cast<scalar_t>(
+              interpolate_aa_single_dim<scalar_t, accscalar_t>(
+                  buffer1, wx, xsize));
+        }
+        odata[n][c][h2][w2] = static_cast<scalar_t>(
+            interpolate_aa_single_dim<scalar_t, accscalar_t>(
+                buffer2, wy, ysize));
+      }
+    }
+  }
+}
+
+template <int interp_size>
+static void upsample_gen2d_out_cuda_template(
+    const Tensor& output,
+    const Tensor& input,
+    IntArrayRef output_size,
+    bool align_corners,
+    c10::optional<double> scales_h,
+    c10::optional<double> scales_w) {
+  TensorArg input_arg{input, "input", 1}, output_arg{output, "output", 2};
+  checkAllSameGPU("upsample_gen2d_out_cuda", {input_arg, output_arg});
+
+  int output_height = output_size[0];
+  int output_width = output_size[1];
+
+  int nbatch = input.size(0);
+  int channels = input.size(1);
+  int input_height = input.size(2);
+  int input_width = input.size(3);
+
+  const int num_kernels = output_height * output_width;
+  const int num_threads = std::min(
+      at::cuda::getCurrentDeviceProperties()->maxThreadsPerBlock, 1024);
+  cudaStream_t stream = at::cuda::getCurrentCUDAStream();
+
+  AT_DISPATCH_FLOATING_TYPES_AND_HALF(
+      input.scalar_type(), "upsample_bilinear2d_out_frame", [&] {
+        using accscalar_t = at::acc_type<scalar_t, true>;
+
+        auto idata = input.packed_accessor64<scalar_t, 4>();
+        auto odata = output.packed_accessor64<scalar_t, 4>();
+
+        const accscalar_t rheight = area_pixel_compute_scale<accscalar_t>(
+            input_height, output_height, align_corners, scales_h);
+        const accscalar_t rwidth = area_pixel_compute_scale<accscalar_t>(
+            input_width, output_width, align_corners, scales_w);
+
+        // We are using static buffer memory of 256 * sizeof(float) per thread
+        // to store weights. Size of weights array is
+        // interp_size = scale * 2 + 1 for bilinear mode
+        TORCH_CHECK(rheight < 127, "Max supported scale factor is 127");
+        TORCH_CHECK(rwidth < 127, "Max supported scale factor is 127");
```

Comment:
Is the max scale factor supported different if we use `bicubic`?

Reply:
Thanks for catching that, yes for bicubic max scale should be at most 63. In general the check should be
```c++
TORCH_CHECK(rheight < 255 / interp_size, "Max supported scale factor is 127 (bilinear), 64 (bicubic)");
```